### PR TITLE
[REF] minor code simplification - remove over-handling of amount comp with zero

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Very minor code simplification

Before
----------------------------------------
Code less readable

After
----------------------------------------
Code more readable

Technical Details
----------------------------------------
This just removes the duplicate assign codes & simplifies the <=0 condition

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contribute/Form/Contribution/Confirm.php [hunk 1]
```diff
@@ -596,20 +596,17 @@ public function buildQuickForm() {
     // the is_monetary concept probably should be too as it can be calculated from
     // the existence of 'amount' & seems fragile.
     if ($this->_contributeMode == 'notify' ||
-      $this->_amount < 0.0 || $this->_params['is_pay_later'] ||
-      ($this->_separateMembershipPayment && $this->_amount <= 0.0)
+      $this->_amount <= 0.0 || $this->_params['is_pay_later']
```
Comment (mattwire) [inline CRM/Contribute/Form/Contribution/Confirm.php]: Worst case we'd end up with the wrong text on a button so this is pretty low risk!  What would we need to do to get rid of `_contributeMode` here?

Also, looks like the is_monetary comment no longer applies and should be removed.  Can the _contributeMode comment get a fixme/todo tag?

File: CRM/Contribute/Form/Contribution/Confirm.php [hunk 2]
```diff
@@ -596,20 +596,17 @@ public function buildQuickForm() {
...
     }
     elseif (!empty($this->_ccid)) {
       $contribButton = ts('Make Payment');
-      $this->assign('button', ts('Make Payment'));
     }
     else {
       $contribButton = ts('Make Contribution');
-      $this->assign('button', ts('Make Contribution'));
     }
+    $this->assign('button', $contribButton);
```
Comment (mattwire) [inline CRM/Contribute/Form/Contribution/Confirm.php]: Yes to these changes.

**Comment by mattwire** [comment]
@eileenmcnaughton This is good to merge pending my comments about the comments.

**Comment by eileenmcnaughton** [comment]
@mattwire I'm gonna merge this as a simplification - I'm not quite sure what to put in the 'comments' as yet because part of my hope is that we can simplify further - ie. can we get away from having 3 different button names?

**Pull request merged**